android/BOINC/app/src/main/java/edu/berkeley/boinc/client/MonitorAsync.kt

```diff
@@ -0,0 +1,345 @@
+package edu.berkeley.boinc.client
+
+import android.graphics.Bitmap
+import android.os.IBinder
+import edu.berkeley.boinc.SplashActivity
+import edu.berkeley.boinc.rpc.*
+import edu.berkeley.boinc.utils.ErrorCodeDescription
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+
+class MonitorAsync(monitor: IMonitor?) : IMonitor {
+    val monitor = monitor!!
+
+    fun quitClientAsync(callback: ((Boolean) -> Unit)? = null) = GlobalScope.async {
+        val result  = quitClient()
+        if (callback != null) {
+            callback(result)
+        }
+        result
+    }
+
+    fun runBenchmarksAsync(callback: ((Boolean) -> Unit)? = null) = GlobalScope.async {
+        val result  = runBenchmarks()
+        if (callback != null) {
+            callback(result)
+        }
+        result
+    }
+
+    fun setGlobalPreferencesAsync(prefs: GlobalPreferences, callback: ((Boolean) -> Unit)? = null) = GlobalScope.async {
+        val result = setGlobalPreferences(prefs)
+        if (callback != null) {
+            callback(result)
+        }
+        result
+    }
+
+    fun setCcConfigAsync(config: String, callback: ((Boolean) -> Unit)? = null) = GlobalScope.async {
+        val result  = setCcConfig(config)
+        if (callback != null) {
+            callback(result)
+        }
+        result
+    }
+
+    override fun asBinder(): IBinder {
+        return monitor.asBinder()
+    }
+
+    override fun attachProject(url: String?, projectName: String?, authenticator: String?): Boolean {
+        return attachProject(url, projectName, authenticator)
```

Comment:
missed `monitor` here, because otherwise you call yourself here

Reply:
fix